File: code_challenges.java
// Fizz Buzz
public class Main {
  static void fizzBuzz() {
    int n = 100;
    for (int i=1; i<n; i++) {
      if (i%15==0){
        System.out.println("FizzBuzz"+" ");
      } else if (i%5==0){
        System.out.println("Buzz"+" ");
      } else if (i%3==0){
        System.out.println("Fizz"+" ");
      } else {
        System.out.println(i);
      }
    }
  }

  public static void main(String[] args) {
    fizzBuzz();
  }
}

// PLUNGE CUT METHOD HINGE CALCULATOR
public class Main {
  public static double hingeWidth(double dbh){
    double hw = dbh*0.10;
    return hw;
  }
  public static double hingeLength(double dbh){
      double hl = dbh*0.80;
      return hl;
  }

  public static String calculate(double dbh) {
    return(String.format("Hinge Width: %s inches \nHinge Length: %s inches", hingeWidth(dbh), hingeLength(dbh)));
  }
    public static void main(String[] args) {
    calculate(22f);
    System.out.println(calculate(22f));
  }
}

// CHECK AGE --- CALCULATE AGE

class Main {
  static int calculateAge(int birthY, int currentY){
    int age = currentY - birthY;
    return age;
  }

  static void checkAge(int birthY, int currentY){
    int age = calculateAge(birthY, currentY);
    if (age < 18){
      System.out.println(age + " —— Access denied —— You are not old enough.");
    } else {
      System.out.println(age + " —— Access granted!");
    }
  }

  public static void main(String[] args){
    checkAge(1956, 2020);
    checkAge(1986, 2020);
    checkAge(1988, 2020);
    checkAge (2010, 2020);
    checkAge(2005, 2020);
  }
}

File: code_challenges.js
// Two Sum
// Given an array of integers, return indices of the two numbers such that they add up to a specific target.
// You may assume that each input would have exactly one solution, and you may not use the same element twice.
//
// Example:
// Given nums = [2, 7, 11, 15], target = 9,
// Because nums[0] + nums[1] = 2 + 7 = 9,
// return [0, 1].

const distinct = (value, index, self) => {
    return self.indexOf(value) === index;
}

var twoSum = function(nums, target) {
    indexes = []
    nums.forEach((number, index) =>{
        nums.forEach((num, i) => {

            if (number + num == target && index != i) {
                indexes.push(index);
                indexes.push(i);
            }
        });

    });
    return(indexes.filter(distinct));
};

// -----------------------------------------------------------------------------


// CHECK AGE // CALCULATE AGE

function checkAge(birthY, currentY) {
  age = calculateAge(birthY, currentY);

  if (age < 18) {
    console.log(`Access Denied —— You are only ${age}.`);
  } else if (age >= 18){
    console.log (`Access Granted! —— Age: ${age}`);
  } else {
    console.log("Your age cannot be accurately determined. Please try again.");
  };
}


function calculateAge(birthY, currentY) {
  age = currentY - birthY;
  return age;
};

checkAge(1986, 2020);
checkAge(2005, 2020);
